From: Daniel Rahn <drahn@suse.de>
Subject: abstract glib API

Glib API may change with minor releases. This patch abstracts glib 
somehow from the various PAN source files to concentrate needed changes
in a single include file.

Adopting to glib 2.16.1 at the same time.

Signed-off-by: Daniel Rahn <drahn@suse.de>
Signed-off-by: Duncan <1i5t5.duncan@cox.net>

--- 
 data-impl/data-impl.cc        |    2 -
 data-impl/data-io.cc          |    3 --
 data-impl/groups.cc           |    2 -
 data-impl/headers.cc          |    3 --
 data-impl/my-tree.cc          |    2 -
 data-impl/profiles.cc         |    3 --
 data-impl/server.cc           |    3 --
 data-impl/xover.cc            |    2 -
 data/article-cache.cc         |    3 --
 data/article-cache.h          |    2 -
 general/e-util.cc             |    3 --
 general/file-util.cc          |    3 --
 general/file-util.h           |   18 ----------------
 general/glib-compat.h         |   45 ++++++++++++++++++++++++++++++++++++++++++
 general/line-reader.cc        |    2 -
 general/locking.h             |    2 -
 general/text-match.cc         |    3 --
 general/time-elapsed.h        |    4 ---
 general/utf8-utils.cc         |    2 -
 general/worker-pool.cc        |    2 -
 general/worker-pool.h         |    3 --
 gui/actions.cc                |    2 -
 gui/body-pane.cc              |    2 -
 gui/dl-headers-ui.cc          |    3 --
 gui/e-charset-picker.c        |    2 -
 gui/group-pane.cc             |    2 -
 gui/group-prefs-dialog.cc     |    2 -
 gui/gui.cc                    |    2 -
 gui/header-pane.cc            |    2 -
 gui/log-ui.cc                 |    2 -
 gui/pan-file-entry.cc         |    2 -
 gui/pan.cc                    |    2 -
 gui/post-ui.cc                |    2 -
 gui/prefs-ui.cc               |    2 -
 gui/prefs.cc                  |    3 --
 gui/profiles-dialog.cc        |    2 -
 gui/render-bytes.cc           |    2 -
 gui/render-bytes.h            |    2 -
 gui/save-ui.cc                |    2 -
 gui/score-add-ui.cc           |    2 -
 gui/score-view-ui.cc          |    2 -
 gui/server-ui.cc              |    3 --
 gui/task-pane.cc              |    3 --
 gui/url.cc                    |    3 --
 tasks/decoder.cc              |    6 ++---
 tasks/nntp-pool.cc            |    2 -
 tasks/nntp.cc                 |    3 --
 tasks/socket-impl-gio.cc      |    3 --
 tasks/task-article.cc         |    2 -
 tasks/task-groups.cc          |    2 -
 tasks/task-post.cc            |    2 -
 tasks/task-xover.cc           |    2 -
 usenet-utils/filter-info.cc   |    3 --
 usenet-utils/gnksa.cc         |    2 -
 usenet-utils/message-check.cc |    2 -
 usenet-utils/mime-utils.h     |    2 -
 usenet-utils/scorefile.cc     |    2 -
 usenet-utils/text-massager.cc |    3 --
 58 files changed, 104 insertions(+), 95 deletions(-)

--- pan-0.132/pan/data/article-cache.cc	2007-08-01 19:00:01.000000000 +0200
+++ BUILD/pan-0.132/pan/data/article-cache.cc	2008-03-29 10:28:09.175341212 +0100
@@ -27,8 +27,7 @@
   #include <unistd.h>
   #include <dirent.h>
 
-  #include <glib.h>
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
   #include <gmime/gmime.h>
 }
 
--- pan-0.132/pan/data/article-cache.h	2007-08-01 19:00:01.000000000 +0200
+++ BUILD/pan-0.132/pan/data/article-cache.h	2008-03-29 10:28:44.560580324 +0100
@@ -23,7 +23,7 @@
 #include <map>
 #include <vector>
 extern "C" {
-  #include <glib/gtypes.h> // for guint64
+  #include <pan/general/glib-compat.h>
 }
 #include <pan/general/string-view.h>
 #include <pan/general/quark.h>
--- pan-0.132/pan/data-impl/data-impl.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/data-impl/data-impl.cc	2008-03-29 10:29:13.585596767 +0100
@@ -23,7 +23,7 @@
 
 #include <config.h>
 extern "C" {
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
 }
 #include <pan/general/debug.h>
 #include <pan/general/file-util.h>
--- pan-0.132/pan/data-impl/data-io.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/data-impl/data-io.cc	2008-03-29 10:29:37.454432677 +0100
@@ -27,8 +27,7 @@
   #include <sys/types.h> // for chmod
   #include <sys/stat.h> // for chmod
   #include <unistd.h>
-  #include <glib.h>
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
 }
 #include <pan/general/debug.h>
 #include <pan/general/file-util.h>
--- pan-0.132/pan/data-impl/groups.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/data-impl/groups.cc	2008-03-29 10:30:03.735353098 +0100
@@ -30,7 +30,7 @@
 
 extern "C" {
   #include <unistd.h>
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
 }
 
 #include <pan/general/debug.h>
--- pan-0.132/pan/data-impl/headers.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/data-impl/headers.cc	2008-03-29 10:30:31.508325815 +0100
@@ -27,8 +27,7 @@
 extern "C" {
   #include <sys/types.h> // for chmod
   #include <sys/stat.h> // for chmod
-  #include <glib/gi18n.h>
-  #include <glib/gmessages.h> // for g_assert
+  #include <pan/general/glib-compat.h>
 }
 #include <pan/general/debug.h>
 #include <pan/general/foreach.h>
--- pan-0.132/pan/data-impl/my-tree.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/data-impl/my-tree.cc	2008-03-29 10:31:02.341402615 +0100
@@ -19,7 +19,7 @@
 
 #include <config.h>
 #include <cassert>
-#include <glib/gmessages.h> // for g_assert
+#include <pan/general/glib-compat.h>
 #include <pan/general/debug.h>
 #include <pan/general/foreach.h>
 #include <pan/general/quark.h>
--- pan-0.132/pan/data-impl/profiles.cc	2007-08-01 19:26:18.000000000 +0200
+++ BUILD/pan-0.132/pan/data-impl/profiles.cc	2008-03-29 10:31:28.674315169 +0100
@@ -26,8 +26,7 @@
 extern "C" {
   #include <sys/types.h> // for chmod
   #include <sys/stat.h> // for chmod
-  #include <glib.h>
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
 }
 #include <pan/general/debug.h>
 #include <pan/general/string-view.h>
--- pan-0.132/pan/data-impl/server.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/data-impl/server.cc	2008-03-29 10:32:01.547454646 +0100
@@ -26,8 +26,7 @@
 #include <set>
 #include <vector>
 extern "C" {
-  #include <glib.h> // for GMarkup
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
 }
 #include <pan/general/debug.h>
 #include <pan/general/file-util.h>
--- pan-0.132/pan/data-impl/xover.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/data-impl/xover.cc	2008-03-29 10:32:27.016337677 +0100
@@ -21,7 +21,7 @@
 #include <cmath>
 #include <fstream>
 extern "C" {
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
   #include <gmime/gmime.h>
 }
 #include <pan/general/debug.h>
--- pan-0.132/pan/general/e-util.cc	2007-08-01 19:00:01.000000000 +0200
+++ BUILD/pan-0.132/pan/general/e-util.cc	2008-03-29 10:24:37.099916303 +0100
@@ -28,8 +28,7 @@
 #include <ctime>
 extern "C" {
   #include <ctype.h>
-  #include <glib.h>
-  #include <glib/gi18n.h>
+  #include "glib-compat.h"
 }
 #include "debug.h"
 #include "e-util.h"
--- pan-0.132/pan/general/file-util.cc	2007-08-01 19:00:01.000000000 +0200
+++ BUILD/pan-0.132/pan/general/file-util.cc	2008-03-29 10:24:53.672496427 +0100
@@ -29,8 +29,7 @@
   #include <sys/stat.h>
   #include <sys/types.h>
 
-  #include <glib.h>
-  #include <glib/gi18n.h>
+  #include "glib-compat.h"
 }
 
 #include "debug.h"
--- pan-0.132/pan/general/file-util.h	2008-03-29 11:48:19.034385493 +0100
+++ BUILD/pan-0.132/pan/general/file-util.h	2008-03-29 11:56:50.316234274 +0100
@@ -28,26 +28,10 @@
 extern "C" {
   #include <stddef.h>
   #include <stdio.h>
-  #include <glib/gtypes.h>
-  #include <glib.h>
+  #include "glib-compat.h"
 }
 #include <pan/general/string-view.h>
 
-#if GLIB_CHECK_VERSION(2,6,0)
-extern "C" {
-  #include <glib/gstdio.h>
-}
-#define g_freopen freopen
-#define g_fopen fopen
-#define g_rmdir rmdir
-#define g_remove remove
-#define g_unlink unlink
-#define g_lstat lstat
-#define g_stat stat
-#define g_rename rename
-#define g_open open
-#endif
-
 namespace pan
 {
   /**
--- pan-0.132/pan/general/glib-compat.h	1970-01-01 01:00:00.000000000 +0100
+++ BUILD/pan-0.132/pan/general/glib-compat.h	2008-04-07 10:24:10.502546064 +0200
@@ -0,0 +1,45 @@
+/* -*- Mode: C++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/*
+ * Pan - A Newsreader for Gtk+
+ * Copyright (C) 2002-2008  Charles Kerr <charles@rebelbase.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#ifndef __G_LIB_H__
+#include <glib.h>
+#endif
+
+#include <glib/gmessages.h>
+#include <glib/gi18n.h>
+#include <glib/gtypes.h>
+#include <glib/gunicode.h>
+#include <glib/gtypes.h>
+#include <glib/gmain.h>
+#include <glib/gtimer.h>
+#include <glib/gthread.h>
+#include <glib/gthreadpool.h>
+
+#if GLIB_CHECK_VERSION(2,6,0)
+   #include <glib/gstdio.h>
+   #define g_freopen freopen
+   #define g_fopen fopen
+   #define g_rmdir rmdir
+   #define g_remove remove
+   #define g_unlink unlink
+   #define g_lstat lstat
+   #define g_stat stat
+   #define g_rename rename
+   #define g_open open
+#endif
--- pan-0.132/pan/general/line-reader.cc	2007-08-01 19:00:01.000000000 +0200
+++ BUILD/pan-0.132/pan/general/line-reader.cc	2008-03-29 10:25:24.401572142 +0100
@@ -1,7 +1,7 @@
 #include <config.h>
 #include <cstdio>
 extern "C"{ 
-  #include <glib/gi18n.h>
+  #include "glib-compat.h"
 }
 #include <cerrno>
 #include "file-util.h"
--- pan-0.132/pan/general/locking.h	2007-08-01 19:00:01.000000000 +0200
+++ BUILD/pan-0.132/pan/general/locking.h	2008-03-29 10:25:38.774075290 +0100
@@ -22,7 +22,7 @@
 #ifndef _Mutex_h_
 #define _Mutex_h_
 
-#include <glib/gthread.h>
+#include "glib-compat.h"
 
 namespace pan
 {
--- pan-0.132/pan/general/text-match.cc	2007-08-01 19:00:01.000000000 +0200
+++ BUILD/pan-0.132/pan/general/text-match.cc	2008-03-29 10:25:55.638665696 +0100
@@ -21,8 +21,7 @@
 extern "C" {
   #include <ctype.h>
   #include <pcre.h>
-  #include <glib.h>
-  #include <glib/gi18n.h>
+  #include "glib-compat.h"
 }
 #include "debug.h"
 #include "log.h"
--- pan-0.132/pan/general/time-elapsed.h	2007-08-01 19:00:01.000000000 +0200
+++ BUILD/pan-0.132/pan/general/time-elapsed.h	2008-03-29 10:26:11.999238473 +0100
@@ -1,9 +1,7 @@
 #ifndef __Time_Elapsed_h__
 #define __Time_Elapsed_h__
 
-#include <glib/gtypes.h> // for GTimeVal
-#include <glib/gmain.h> // for g_get_current_time
-#include <glib/gtimer.h> // for GUSEC_PER_SEC
+#include "glib-compat.h" 
 
 namespace pan
 {
--- pan-0.132/pan/general/utf8-utils.cc	2007-08-01 19:00:01.000000000 +0200
+++ BUILD/pan-0.132/pan/general/utf8-utils.cc	2008-03-29 10:26:32.335950476 +0100
@@ -23,7 +23,7 @@
 #include <string.h>
 #include <glib.h>
 extern "C" {
-#include <glib/gi18n.h>
+#include "glib-compat.h"
 }
 #include <locale.h>
 #include <gmime/gmime.h>
--- pan-0.132/pan/general/worker-pool.cc	2007-08-01 19:00:01.000000000 +0200
+++ BUILD/pan-0.132/pan/general/worker-pool.cc	2008-03-29 10:26:49.336545697 +0100
@@ -21,7 +21,7 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 #include <cassert>
-#include <glib/gmain.h> // for g_idle_add
+#include "glib-compat.h"
 #include <pan/general/foreach.h>
 #include "worker-pool.h"
 
--- pan-0.132/pan/general/worker-pool.h	2007-08-01 19:00:01.000000000 +0200
+++ BUILD/pan-0.132/pan/general/worker-pool.h	2008-03-29 10:27:08.833228330 +0100
@@ -24,8 +24,7 @@
 #define _Worker_Pool_H_
 
 #include <set>
-#include <glib/gtypes.h>
-#include <glib/gthreadpool.h>
+#include "glib-compat.h"
 
 namespace pan
 {
--- pan-0.132/pan/gui/actions.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/gui/actions.cc	2008-03-29 10:33:13.569952196 +0100
@@ -19,7 +19,7 @@
 
 #include <config.h>
 extern "C" {
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
   #include <gtk/gtk.h>
 }
 #include <pan/general/debug.h>
--- pan-0.132/pan/gui/body-pane.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/gui/body-pane.cc	2008-03-29 10:33:57.167464724 +0100
@@ -22,7 +22,7 @@
 #include <cmath>
 #include <iostream>
 extern "C" {
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
   #include <gtk/gtk.h>
   #include <gdk/gdkkeysyms.h>
   #include <gmime/gmime.h>
--- pan-0.132/pan/gui/dl-headers-ui.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/gui/dl-headers-ui.cc	2008-03-29 10:34:25.684454334 +0100
@@ -19,8 +19,7 @@
 
 #include <config.h>
 extern "C" {
-  #include <glib.h>
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
   #include <gtk/gtk.h>
 }
 #include <pan/general/foreach.h>
--- pan-0.132/pan/gui/e-charset-picker.c	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/gui/e-charset-picker.c	2008-03-29 11:40:38.814395583 +0100
@@ -37,7 +37,7 @@
 #include <gtk/gtksignal.h>
 #include <gtk/gtkmessagedialog.h>
 
-#include <glib/gi18n.h>
+#include <pan/general/glib-compat.h>
 
 typedef enum {
 	E_CHARSET_UNKNOWN,
--- pan-0.132/pan/gui/group-pane.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/gui/group-pane.cc	2008-03-29 10:34:50.621319878 +0100
@@ -21,7 +21,7 @@
 #include <cassert>
 #include <deque>
 extern "C" {
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
   #include <gtk/gtk.h>
 }
 #include <pan/general/debug.h>
--- pan-0.132/pan/gui/group-prefs-dialog.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/gui/group-prefs-dialog.cc	2008-03-29 10:35:20.810367936 +0100
@@ -19,7 +19,7 @@
 
 #include <config.h>
 extern "C" {
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
   #include <gtk/gtk.h>
 }
 #include <pan/general/debug.h>
--- pan-0.132/pan/gui/gui.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/gui/gui.cc	2008-03-29 10:35:39.707024080 +0100
@@ -24,7 +24,7 @@
 extern "C" {
   #include <sys/types.h> // for chmod
   #include <sys/stat.h> // for chmod
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
 }
 #include <pan/general/debug.h>
 #include <pan/general/e-util.h>
--- pan-0.132/pan/gui/header-pane.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/gui/header-pane.cc	2008-03-29 10:35:58.419673922 +0100
@@ -19,7 +19,7 @@
 
 extern "C" {
   #include <config.h>
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
   #include <gtk/gtk.h>
 }
 #include <cctype>
--- pan-0.132/pan/gui/log-ui.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/gui/log-ui.cc	2008-03-29 10:36:20.216430978 +0100
@@ -21,7 +21,7 @@
 #include <ostream>
 #include <fstream>
 extern "C" {
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
   #include <gtk/gtk.h>
 }
 #include <pan/general/foreach.h>
--- pan-0.132/pan/gui/pan.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/gui/pan.cc	2008-03-29 10:36:38.817077115 +0100
@@ -20,7 +20,7 @@
 #include <config.h>
 #include <signal.h>
 extern "C" {
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
   #include <gtk/gtk.h>
   #include <gmime/gmime.h>
 }
--- pan-0.132/pan/gui/pan-file-entry.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/gui/pan-file-entry.cc	2008-03-29 11:38:02.264968572 +0100
@@ -20,7 +20,7 @@
 #include <config.h>
 #include <string>
 extern "C" {
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
   #include <gtk/gtk.h>
 }
 #include <pan/general/file-util.h>
--- pan-0.132/pan/gui/post-ui.cc	2007-08-01 19:27:37.000000000 +0200
+++ BUILD/pan-0.132/pan/gui/post-ui.cc	2008-03-29 10:37:20.826536722 +0100
@@ -23,7 +23,7 @@
 #include <sstream>
 extern "C" {
   #include <gmime/gmime.h>
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
   #include <gtk/gtk.h>
 #ifdef HAVE_GTKSPELL
   #include <gtkspell/gtkspell.h>
--- pan-0.132/pan/gui/prefs.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/gui/prefs.cc	2008-03-29 10:37:40.143208016 +0100
@@ -23,8 +23,7 @@
 #include <sstream>
 #include <string>
 extern "C" {
-  #include <glib.h>
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
 }
 #include <pan/general/debug.h>
 #include <pan/general/foreach.h>
--- pan-0.132/pan/gui/prefs-ui.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/gui/prefs-ui.cc	2008-03-29 10:38:00.299908596 +0100
@@ -19,7 +19,7 @@
 
 #include <config.h>
 extern "C" {
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
   #include <gtk/gtk.h>
 }
 #include <pan/general/debug.h>
--- pan-0.132/pan/gui/profiles-dialog.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/gui/profiles-dialog.cc	2008-03-29 10:38:21.064630409 +0100
@@ -24,7 +24,7 @@
 #include <config.h>
 #include <iostream>
 extern "C" {
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
   #include <gtk/gtk.h>
 }
 #include <pan/general/debug.h>
--- pan-0.132/pan/gui/render-bytes.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/gui/render-bytes.cc	2008-03-29 10:38:44.165433544 +0100
@@ -1,4 +1,4 @@
-#include <glib/gutils.h> // g_snprintf
+#include <pan/general/glib-compat.h>
 #include "render-bytes.h"
 
 namespace pan
--- pan-0.132/pan/gui/render-bytes.h	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/gui/render-bytes.h	2008-03-29 10:41:52.299965125 +0100
@@ -1,6 +1,6 @@
 #ifndef RENDER_BYTES_H
 #define RENDER_BYTES_H
-#include <glib/gtypes.h>
+#include <pan/general/glib-compat.h>
 namespace pan {
   extern char* render_bytes (guint64);
 }
--- pan-0.132/pan/gui/save-ui.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/gui/save-ui.cc	2008-03-29 10:39:04.934155706 +0100
@@ -19,7 +19,7 @@
 
 #include <config.h>
 extern "C" {
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
   #include <gtk/gtk.h>
 }
 #include <pan/general/debug.h>
--- pan-0.132/pan/gui/score-add-ui.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/gui/score-add-ui.cc	2008-03-29 10:39:24.446834281 +0100
@@ -20,7 +20,7 @@
 #include <config.h>
 #include <cassert>
 extern "C" {
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
 }
 #include <pan/general/debug.h>
 #include <pan/general/text-match.h>
--- pan-0.132/pan/gui/score-view-ui.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/gui/score-view-ui.cc	2008-03-29 10:39:43.603498787 +0100
@@ -19,7 +19,7 @@
 
 #include <config.h>
 extern "C" {
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
   #include <gtk/gtk.h>
 }
 #include <pan/general/debug.h>
--- pan-0.132/pan/gui/server-ui.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/gui/server-ui.cc	2008-03-29 10:40:05.944273805 +0100
@@ -21,8 +21,7 @@
 #include <cstdlib>
 #include <cstring>
 extern "C" {
-  #include <glib.h>
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
   #include <gtk/gtk.h>
 }
 #include <pan/general/foreach.h>
--- pan-0.132/pan/gui/task-pane.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/gui/task-pane.cc	2008-03-29 10:40:30.369121269 +0100
@@ -19,8 +19,7 @@
 
 #include <config.h>
 extern "C" {
-  #include <glib.h>
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
   #include <gtk/gtk.h>
 }
 #include <pan/general/debug.h>
--- pan-0.132/pan/gui/url.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/gui/url.cc	2008-03-29 10:40:57.562064958 +0100
@@ -21,8 +21,7 @@
 #include <iostream>
 #include <string>
 extern "C" {
-  #include <glib.h>
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
 };
 #include <pan/general/log.h>
 #include "url.h"
--- pan-0.132/pan/tasks/decoder.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/tasks/decoder.cc	2008-03-29 10:42:47.169870620 +0100
@@ -27,9 +27,9 @@
 #include <ostream>
 #include <fstream>
 extern "C" {
-#  define PROTOTYPES
-#  include <uulib/uudeview.h>
-#  include <glib/gi18n.h>
+	#define PROTOTYPES
+	#include <uulib/uudeview.h>
+	#include <pan/general/glib-compat.h>
 };
 #include <pan/general/debug.h>
 #include <pan/general/file-util.h>
--- pan-0.132/pan/tasks/nntp.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/tasks/nntp.cc	2008-03-29 10:43:05.866520075 +0100
@@ -21,8 +21,7 @@
 #include <cassert>
 #include <cstdarg>
 extern "C" {
-  #include <glib.h>
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
 }
 #include <pan/general/debug.h>
 #include <pan/general/log.h>
--- pan-0.132/pan/tasks/nntp-pool.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/tasks/nntp-pool.cc	2008-03-29 10:43:46.291920677 +0100
@@ -19,7 +19,7 @@
 
 #include <config.h>
 #include <ctime>
-#include <glib/gi18n.h>
+#include <pan/general/glib-compat.h>
 #include <pan/general/debug.h>
 #include <pan/general/foreach.h>
 #include <pan/general/log.h>
--- pan-0.132/pan/tasks/socket-impl-gio.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/tasks/socket-impl-gio.cc	2008-03-29 10:44:17.172975084 +0100
@@ -31,8 +31,7 @@
 
 extern "C" {
   #include <unistd.h>
-  #include <glib.h>
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
 }
 
 #include <pan/general/file-util.h>
--- pan-0.132/pan/tasks/task-article.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/tasks/task-article.cc	2008-03-29 10:44:38.725711567 +0100
@@ -25,7 +25,7 @@
 #include <algorithm>
 #include <cassert>
 extern "C" {
-#include <glib/gi18n.h>
+#include <pan/general/glib-compat.h>
 }
 #include <pan/general/debug.h>
 #include <pan/general/file-util.h>
--- pan-0.132/pan/tasks/task-groups.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/tasks/task-groups.cc	2008-03-29 10:44:59.634426490 +0100
@@ -20,7 +20,7 @@
 #include <config.h>
 #include <cassert>
 extern "C" {
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
 }
 #include <pan/general/debug.h>
 #include <pan/general/foreach.h>
--- pan-0.132/pan/tasks/task-post.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/tasks/task-post.cc	2008-03-29 10:45:21.387170729 +0100
@@ -19,7 +19,7 @@
 
 #include <config.h>
 extern "C" {
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
 }
 #include "task-post.h"
 #include <pan/general/debug.h>
--- pan-0.132/pan/tasks/task-xover.cc	2007-08-01 19:00:00.000000000 +0200
+++ BUILD/pan-0.132/pan/tasks/task-xover.cc	2008-03-29 10:45:47.616068719 +0100
@@ -21,7 +21,7 @@
 #include <cassert>
 #include <cerrno>
 extern "C" {
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
   #include <gmime/gmime-utils.h>
 }
 #include <pan/general/debug.h>
--- pan-0.132/pan/usenet-utils/filter-info.cc	2007-08-01 18:59:59.000000000 +0200
+++ BUILD/pan-0.132/pan/usenet-utils/filter-info.cc	2008-03-29 10:46:25.993383793 +0100
@@ -19,8 +19,7 @@
 
 #include <config.h>
 extern "C" {
-  #include <glib.h>
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
 }
 #include <pan/general/foreach.h>
 #include "filter-info.h"
--- pan-0.132/pan/usenet-utils/gnksa.cc	2007-08-01 18:59:59.000000000 +0200
+++ BUILD/pan-0.132/pan/usenet-utils/gnksa.cc	2008-03-29 10:46:59.906547006 +0100
@@ -36,7 +36,7 @@
 {
   #include <ctype.h>
   #include <unistd.h>
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
 }
 
 #include <pan/general/debug.h>
--- pan-0.132/pan/usenet-utils/message-check.cc	2007-08-01 18:59:59.000000000 +0200
+++ BUILD/pan-0.132/pan/usenet-utils/message-check.cc	2008-03-29 10:47:20.727261653 +0100
@@ -22,7 +22,7 @@
 #include <string.h>
 #include <glib.h>
 extern "C" {
-#include <glib/gi18n.h>
+#include <pan/general/glib-compat.h>
 }
 #include <pan/general/debug.h>
 #include <pan/general/foreach.h>
--- pan-0.132/pan/usenet-utils/mime-utils.h	2007-08-01 18:59:59.000000000 +0200
+++ BUILD/pan-0.132/pan/usenet-utils/mime-utils.h	2008-03-29 10:48:34.309790222 +0100
@@ -21,7 +21,7 @@
 #define _UtilMime_h_
 
 #include <vector>
-#include <glib/gtypes.h>
+#include <pan/general/glib-compat.h>
 #include <gmime/gmime-filter.h>
 #include <gmime/gmime-stream.h>
 #include <gmime/gmime-message.h>
--- pan-0.132/pan/usenet-utils/scorefile.cc	2007-08-01 18:59:59.000000000 +0200
+++ BUILD/pan-0.132/pan/usenet-utils/scorefile.cc	2008-03-29 10:47:40.255932291 +0100
@@ -22,7 +22,7 @@
 #include <sstream>
 #include <string>
 extern "C" {
-  #include <glib/gi18n.h>
+  #include <pan/general/glib-compat.h>
 }
 #include <pan/general/debug.h>
 #include <pan/general/log.h>
--- pan-0.132/pan/usenet-utils/text-massager.cc	2007-08-01 18:59:59.000000000 +0200
+++ BUILD/pan-0.132/pan/usenet-utils/text-massager.cc	2008-03-29 10:48:11.144993711 +0100
@@ -20,9 +20,8 @@
 #include <config.h>
 #include <vector>
 extern "C" {
-#include <glib/gi18n.h>
+#include <pan/general/glib-compat.h>
 }
-#include <glib/gunicode.h>
 #include "text-massager.h"
 
 using namespace pan;
